### ISSUE TITLE
Add Podman support

### ISSUE THREAD

**Issue by rhatdan** [inspec/train#525]
I was told this was in the wrong place.

https://github.com/inspec/inspec/issues/4299

**Comment by kekaichinose** [comment]
@rhatdan can you please detail the entire ask here? What is Podman? Why would you need it? How do you envision using it with Chef InSpec?

**Comment by kekaichinose** [comment]
@rhatdan please give us more info on this ask. What is Podman? How do you envision using it with Chef InSpec?

**Comment by rhatdan** [comment]
[Podman](https://podman.io/) is a replacement for the Docker CLI.  The Docker CLI has been dropped from the default distributions on RHEL8/Centos8 Systems.

From a quick look at your code, the first step would be to get ruby support for Podman.

**Comment by ziporah** [comment]
Just as feedback for those on the search of finding a solution running inspec for podman.
The trick is running the podman system service and then use the docker api that is exposed by the service.
I hope this saves some people the days of search I've gone trough to find this solution instead of trying to find a train modification that supports podman/containerd

```
podman system service --time=0 unix:///tmp/podman.sock & DOCKER_SOCK_PID="$!"
export DOCKER_HOST=unix:///tmp/podman.sock
export CONTAINER_TAG=local-image:blah
docker build -t ${CONTAINER_TAG} .
CID=$(docker run --rm -d -i --init --entrypoint=/bin/cat ${CONTAINER_TAG})
inspec exec spec/docker_spec.rb --show-progress -t docker://${CID}
docker stop ${CID}
kill -9 ${DOCKER_SOCK_PID}
```

**Comment by rhatdan** [comment]
Care to write a blog on how you did this and its benefits?

**Comment by ziporah** [comment]
I'll add this to my personal planning, update here when I have something usefull, some small howto or small poc.

**Comment by ziporah** [comment]
well and [there](https://ziporah.github.io/2021-01-13-test-podman-with-inspec/) you have it. Should be easy enough for people to understand and it has a working example in it.

**Comment by damnhandy** [comment]
@ziporah This is brilliant. The blog post is incredibly helpful here. We have been extremely happy with Podman & Buildah, but the ability to test the builds was a bit of a challenge. We considered alternatives to InSpec/CINC Auditor, but nothing else comes close. This is a step in the right direction. 

@kekaichinose I can give you some details on how I'd want to InSpec to support Podman, or any tooling based on the [containers/image](https://github.com/containers/image) library for that matter. We're using the Docker train to verify the structure of our containers. This of course means spinning up a container post-build in order to test it. Right now, it has to use docker as the Gem behind the train relies on the Docker Socket API. Ideally, on would be able to verify the container contents of an image by either using one of the transports that backs the [containers/image](https://github.com/containers/image/blob/master/docs/containers-transports.5.md) library. This could more flexible and possibly not have to rely on spinning up an a container to validate the build structure. Tools like [Buildah](https://buildah.io) and [Podman](https://podman.io) offer some interesting opportunities that Docker does not support at this time.

**Comment by sharpyy** [comment]
+1

**Comment by ziporah** [comment]
It has been a while, but I believe adding this script in the path of the container did the trick:
```
08:29 $ cat docker
#!/bin/sh
exec /usr/bin/podman -r "$@"
```

**Comment by Vasu1105** [comment]
For now, we won't be able to support Podman as there is no ruby library that we can use to fetch the Podman API's.
 
The workaround to use InSpec for Podman can be to use the systemd to activate the Podman socket.
and Enable programs using Docker to interact with the rootless Podman socket by setting DOCKER_HOST environment variable as mentioned in this doc https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/building_running_and_managing_containers/assembly_using-the-container-tools-api_using-the-container-tools-cli
 
So steps that worked for me and which I have tried is
 
1) I created the vagrant centos8 machine installed the Podman on it
2) installed the podman-remote package using `yum install podman-remore` command
3) Activated the podman socket using the `systemctl --user enable --now podman.socket` command
4) set the DOCKER_HOST environment variable using `export DOCKER_HOST=unix:///run/user/<uid>/podman//podman.sock`
5) Installed the InSpec on the VM where the Podman is running
and then I ran the below commands which are mentioned by @zipoah in his blog.
```
export CT=alpine:latest
CID=$(podman run --rm -d -i --init --entrypoint=/bin/cat ${CT})
inspec exec linux-baseline/ -t docker://${CID}
```
 
This all is explored on Centos-8 system.
 
I also tried installing the `podman-docker` package to check whether I will be able to use the docker resources.
I used all the same steps, only instead of installing the Podman I installed the Podman-docker so that I can emulate docker commands using Podman, as I felt that doing this I will able to use the docker resources like docker_image and docker_container but that did not work as it's unable to parse the data correctly.
 
So looking at overall things we currently feel people can use the workaround to work with InSpec. And wait for the podman ruby library supporting Podman API's
